omod/src/main/java/org/openmrs/module/appointments/web/controller/AppointmentController.java

```diff
@@ -191,4 +192,33 @@ public ResponseEntity<Object> rescheduleAppointment(@PathVariable("uuid") String
         }
     }
 
+    /**
+     * Returns a list of appointment on a date and status
+     * @param forDate the appointment date
+     * @param status - the appointment status i.e. scheduled, cancelled, completed,
+     * @return list
+     */
+    @RequestMapping(method = RequestMethod.GET, value = "appointmentStatus")
+    @ResponseBody
+    public ResponseEntity<Object> getAppointmentByDateAndStatus(@RequestParam(value = "forDate") String forDate, @RequestParam(value = "status") String status)  {
+        try {
+            if(StringUtils.isEmpty(forDate) || StringUtils.isEmpty(status)) {
+                return new ResponseEntity<>("The request requires appointment date and status", HttpStatus.BAD_REQUEST);
+            }
+            // Parse the forDate string into a Date object
+            Date date = DateUtil.convertToLocalDateFromUTC(forDate);
+            // Parse the status string into an AppointmentStatus enum
+            AppointmentStatus appointmentStatus;
+            try {
+                appointmentStatus = AppointmentStatus.valueOf(status);
+            } catch (IllegalArgumentException e) {
+                return new ResponseEntity<>("Invalid status value", HttpStatus.BAD_REQUEST);
+            }
+            List<Appointment> appointments = appointmentsService.getAllAppointments(date, appointmentStatus);
+            return new ResponseEntity<>(appointmentMapper.constructResponse(appointments), HttpStatus.OK);
+        } catch (Exception e) {
+            log.error("Runtime error while trying to fetch appointments by status and date", e);
+            return new ResponseEntity<>(RestUtil.wrapErrorResponse(e, e.getMessage()), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
```

Comment:
I don't think this REST endpoint is what we want.  Having a new endpoint called "appointmentStatus" does not feel like the right verbage.  What you really want to do (I think) is just to use the existing "all" endpoint, but modify it to filter on status as well as on date (much like you did in the service and dao layers).  So, can't we - instead of creating a new endpoint here - simply modify the `getAllAppointments` method that already exists to take in an optional `status` request parameter, parses this status parameter into an enum (as you are doing above) and then call the new service method you just created that takes in both the date and the status?